thirdeye-core/src/main/java/ai/startree/thirdeye/alert/AlertDetectionIntervalCalculator.java

```diff
@@ -69,33 +68,31 @@ public Interval getCorrectedInterval(final AlertApi alertApi, final long taskSta
     return getCorrectedInterval(alertId,
         taskStartMillis,
         taskEndMillis,
-        templateWithProperties);
+        templateWithProperties, applyMutabilityPeriod);
   }
 
   /**
    * Applies delay and granularity flooring to the task start and end.
-   *
-   *
    * Notes:
    * Assumes that the caller of this method is setting the corrected endTime as the alert
    * lastTimestamp once the alert pipeline has run successfully.
    */
   public Interval getCorrectedInterval(final AlertDTO alertDTO, final long taskStartMillis,
-      final long taskEndMillis) throws IOException, ClassNotFoundException {
+      final long taskEndMillis, final boolean applyMutabilityPeriod) throws IOException, ClassNotFoundException {
     // render properties - startTime/endTime not important - objective is to get metadata
     final AlertTemplateDTO templateWithProperties = alertTemplateRenderer.renderAlert(alertDTO,
         DUMMY_INTERVAL);
 
     return getCorrectedInterval(alertDTO.getId(),
         taskStartMillis,
         taskEndMillis,
-        templateWithProperties);
+        templateWithProperties,
+        applyMutabilityPeriod);
```

Comment:
good point. I think had to maintain 2 variable to do so, and returning 2 variables in java is somewhat annoying because it requires to introduce an object. 
Will see what I can do.

Reply:
Oh ya, I always forget you can only return one variable. 😅